OWL2/java/de/unibremen/informatik/OWL2Parser.java

```diff
@@ -45,20 +48,87 @@
 	private static OWLOntologyIRIMapperImpl mapper =
 		new OWLOntologyIRIMapperImpl();
 	private static URI uri = null;
+	private static int maxRedirects = 0;
+
+	private static void checkMaxRedirects() {
+		String t = System.getProperty("http.maxRedirects", null);
+		if (t == null || t.isEmpty()) {
+			maxRedirects = 20;
+			return;
+		}
+		try {
+			int n = Integer.parseInt(t, 10);
+			if (n > 0) {
+				maxRedirects = n;
+			}
+		} catch (NumberFormatException nfe) {
+			System.err.println("Ignoring invalid system property "
+				+ "'http.maxRedirects' - invalid number '" + t + "'");
+		}
+	}
+
+	private static URLConnection getConnection(URL url, int redirCount)
+		throws IOException
+	{
+		URLConnection c = url.openConnection();
+		if (!(c instanceof HttpURLConnection)) {
+			return c;
+		}
+		HttpURLConnection con = (HttpURLConnection) c;
+		con.addRequestProperty("Accept", "text/plain");
+		con.setConnectTimeout(300 * 1000);	// 5 min timeout
```

Comment:
couldn't this timeout lead to problems for very large ontologies that take longer than 5 min?